server/cache.go

```diff
@@ -336,6 +348,18 @@ func (c *clusterInfo) putStoreLocked(store *storeInfo) error {
 	return nil
 }
 
+func (c *clusterInfo) blockStore(storeID uint64) {
+	c.Lock()
+	defer c.Unlock()
```

Comment:
No, if we have some logic like `if err; return err`, we may use `defer`, but here if we can make sure we can't have no return, we should not use `defer`.